### DISCUSSION TITLE
How to select multiple commits filtered by filename for rebase?

### DISCUSSION THREAD

**Discussion by kik4444** [jesseduffield/lazygit#3540] [Q&A]
Recently I had to rebase multiple commits to erase some passwords that had mistakenly been pushed to a repo. I went to the commits screen, hit Ctrl+s to filter by file name and found the commits related to the file in question. But afterwards I couldn't figure out how to pick the commits for `edit` during a rebase. The `e` shortcut showed an error message that `edit` is not available when filtering by name, but if I exit the filter lazygit does not jump to the commit I was hovering on before. And the commits I had to rebase were not ordered one after the other so I couldn't range select them all at once.

In the end I settled for screenshotting the commit sha-sums when filtering by filename, then manually going to the oldest commit by hitting Ctrl+/ and searching for its sha-sum, starting a rebase from there, then finding the remaining commits in the same manner and marking them as `edit` with `e`.

Is there a simpler way I could have done this? I also tried starting a rebase from the first commit, but then I could not find the file in question by filtering for it because it was not yet created at the time of the first commit, and even if it was I cannot mark a commit for `edit`ing while I'm in the `filter-by-name` view anyway. Did I miss something, or is this an area for future improvement?

**Comment by mark2185** [comment]
Not necessarily an answer you were looking for, but I usually use [this](https://rtyley.github.io/bfg-repo-cleaner/) for such matters.

**Comment by stefanhaller** [comment]
Yes, this should be improved; it's related to #2554 (not the issue itself, but the discussion it contains). However, that discussion is only about drop, fixup, reword, and things like that; it doesn't talk about edit, and there are several questions related to enabling edit in filter mode. Also, enabling range-select in filter mode is non-trivial, technically. So, not an easy thing to fix, but could be fun to think about all the challenges. I have several other things on my todo list though, so I won't work in this any time soon.

> but if I exit the filter lazygit does not jump to the commit I was hovering on before.

This is fixed on master (see #3416), so this might help a little bit at least if you can build from source.
